clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java

```diff
@@ -84,6 +84,24 @@ private GravitinoClient(
     this.metalake = loadMetalake(metalakeName);
   }
 
+  /**
+   * Constructs a new GravitinoClient with the given URI, authenticator and AuthDataProvider.
+   *
+   * @param uri The base URI for the Gravitino API.
+   * @param authDataProvider The provider of the data which is used for authentication.
+   * @param checkVersion Whether to check the version of the Gravitino server. Gravitino does not
+   *     support the case that the client-side version is higher than the server-side version.
+   * @param headers The base header for Gravitino API.
+   * @throws NoSuchMetalakeException if the metalake with specified name does not exist.
+   */
```

Comment:
How is the metalake set after calling this constructor?

Reply:
Hello, I will close this PR first and make some modifications. Later, I will open a new PR for further discussion